python/cudf_polars/cudf_polars/dsl/expressions/string.py

```diff
@@ -760,12 +760,30 @@ def do_evaluate(
             # TODO: ignores ambiguous
             format, strict, _, _ = self.options
             col = self.children[0].evaluate(df, context=context)
+            plc_col = col.obj
+            if format is None:
+                # Polars begins inference with the first non null value
+                int_range = plc.column.Column.from_iterable_of_py(range(col.size))
+                boolmask = plc.unary.is_valid(plc_col)
+
+                table = plc.stream_compaction.apply_boolean_mask(
+                    plc.Table([int_range]), boolmask
+                )
```

Comment:
Would be nice to have a short circuiting case for `if col.null_count == 0`, so we don't need to do more work finding the first non null element.